### PR TITLE
DEV-2355: Tablet and grabbing for hand trackers

### PR DESCRIPTION
**Pull request by sethalves**
This PR makes changes to the grabbing and tablet modules which allow camera-tracked hands to use the various interfaces.

https://highfidelity.atlassian.net/browse/DEV-2355

- re-enable prefer finger over stylus
- allow thumb+index finger pinch to mean same as trigger click
- touching tips of index-fingers together means walk forward
- get near-grab working with camera-tracked hands
- if hands are tracked, make mini-tablet be just a big open-tablet button
- disable far-trigger and grab if using hand tracker
- don't allow the hand with the mini-tablet to trigger opening the tablet
- use same near-grab module to grab entities and overlays
- fix angularVelocity reported by leap-motion plugin
- touch tips of index fingers to walk forward, tips of thumbs to walk back
- double OK to open/close tablet

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21755/HighFidelity-Beta-PR16302-d2d7cfc.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21755/HighFidelity-Quest-Beta-PR16302-d2d7cfc.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21755/HighFidelity-Beta-PR16302-d2d7cfc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21755/HighFidelity-Beta-Interface-PR16302-d2d7cfc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21755/HighFidelity-Beta-PR16302-d2d7cfc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21755/HighFidelity-Beta-Interface-PR16302-d2d7cfc.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21792/HighFidelity-Beta-PR16302-35fa6f3.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21792/HighFidelity-Quest-Beta-PR16302-35fa6f3.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21792/HighFidelity-Beta-PR16302-35fa6f3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21792/HighFidelity-Beta-Interface-PR16302-35fa6f3.dmg

**Comment by sethalves** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21802/HighFidelity-Beta-PR16302-35fa6f3.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21802/HighFidelity-Quest-Beta-PR16302-35fa6f3.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21802/HighFidelity-Beta-PR16302-35fa6f3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21802/HighFidelity-Beta-Interface-PR16302-35fa6f3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21927/HighFidelity-Beta-PR16302-fe4c6cb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21927/HighFidelity-Beta-Interface-PR16302-fe4c6cb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21927/HighFidelity-Beta-PR16302-fe4c6cb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21927/HighFidelity-Beta-Interface-PR16302-fe4c6cb.exe

**Comment by sethalves** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21957/HighFidelity-Beta-PR16302-fe4c6cb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21957/HighFidelity-Beta-Interface-PR16302-fe4c6cb.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21957/HighFidelity-Beta-PR16302-fe4c6cb.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21957/HighFidelity-Quest-Beta-PR16302-fe4c6cb.apk)

**Comment by sethalves** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21964/HighFidelity-Beta-PR16302-fe4c6cb.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21964/HighFidelity-Quest-Beta-PR16302-fe4c6cb.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21964/HighFidelity-Beta-PR16302-fe4c6cb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21964/HighFidelity-Beta-Interface-PR16302-fe4c6cb.dmg

**Comment by sethalves** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21966/HighFidelity-Beta-PR16302-fe4c6cb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21966/HighFidelity-Beta-Interface-PR16302-fe4c6cb.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21966/HighFidelity-Beta-PR16302-fe4c6cb.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21966/HighFidelity-Quest-Beta-PR16302-fe4c6cb.apk)

**Comment by sethalves** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21979/HighFidelity-Beta-PR16302-fe4c6cb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21979/HighFidelity-Beta-Interface-PR16302-fe4c6cb.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21979/HighFidelity-Beta-PR16302-fe4c6cb.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21979/HighFidelity-Quest-Beta-PR16302-fe4c6cb.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21979/HighFidelity-Beta-PR16302-fe4c6cb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21979/HighFidelity-Beta-Interface-PR16302-fe4c6cb.exe

File: scripts/system/controllers/controllerModules/trackedHandWalk.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            }
+
+            if (this.leftThumbPos && this.rightThumbPos) {
+                var thumbTipDistance = Vec3.distance(this.leftThumbPos, this.rightThumbPos);
+                if (thumbTipDistance < this.touchOnBelowDistance) {
+                    this.walkingBackward = true;
+                    this.controlPoint = this.getControlPoint();
+                } else if (this.walkingBackward && thumbTipDistance > this.touchOffAboveDistance) {
+                    this.walkingBackward = false;
+                }
```
Comment (AndrewMeadows) [inline scripts/system/controllers/controllerModules/trackedHandWalk.js]: No else case for `this.walkingBackward`.

File: scripts/system/controllers/controllerModules/trackedHandWalk.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+            }
+
+            if ((this.walkingForward || this.walkingBackward) && !this.mappingEnabled) {
+                Controller.enableMapping(this.mappingName);
+                this.mappingEnabled = true;
+            } else if (!(this.walkingForward || this.walkingBackward) && this.mappingEnabled) {
+                this.inputMapping.disable();
+                this.mappingEnabled = false;
+            }
```
Comment (AndrewMeadows) [inline scripts/system/controllers/controllerModules/trackedHandWalk.js]: No `else` case for `this.mappingEnabled`.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22603/HighFidelity-Beta-PR16302-6a911f4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22603/HighFidelity-Beta-Interface-PR16302-6a911f4.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22603/HighFidelity-Beta-PR16302-6a911f4.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22603/HighFidelity-Quest-Beta-PR16302-6a911f4.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22602/HighFidelity-Beta-PR16302-7596c96.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22602/HighFidelity-Quest-Beta-PR16302-7596c96.apk)

**Comment by sethalves** [comment]
closing in favor of https://github.com/highfidelity/hifi/pull/16125/files

**Pull request closed without merging**